src/fairseq2/recipes/asr/_common.py

```diff
@@ -178,6 +180,16 @@ def update_batch_metrics(self, batch: Seq2SeqBatch) -> None:
             self.total_num_examples.update(num_examples)
             self.total_num_elements.update(num_elements)
 
+    @torch.inference_mode()
+    def update_extra_metrics(
```

Comment:
should it be generic ?

Reply:
we should have it already in fs2 main somewhere

Reply:
Does not need to be generic because we don't have generic recipes anymore.

Will add this function and the WER / BLEU metric to the open-source ASR recipe though.